cf.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"code.cloudfoundry.org/cli/plugin"
+)
+
+type Cf struct {
+	Cli     plugin.CliConnection
+	Verbose bool
+}
+
+func (c *Cf) generateEmptyApp() (string, error) {
+	// Create a temporary directory containing a minimal staticfile-buildpack app.
+	tempAppDir, err := ioutil.TempDir("", "tmp-tunnel-")
+	if err != nil {
+		return "", err
+	}
+	staticfilePath := filepath.Join(tempAppDir, "Staticfile")
+	err = ioutil.WriteFile(staticfilePath, []byte(""), 0644)
+	if err != nil {
+		os.RemoveAll(tempAppDir)
+		return "", err
+	}
+	return tempAppDir, nil
+}
+
+func (c *Cf) pushAppWithoutRoute(appName string, appDir string) error {
+	return c.run("push", appName,
+		"-p", appDir,
+		"-b", "staticfile_buildpack",
+		"-m", "64M",
+		"-k", "64M",
+		"-i", "1",
+		"--health-check-type", "none",
+		"--no-route",
+		"--no-manifest",
+	)
+}
+
+func (c *Cf) getAppGUID(appName string) (string, error) {
+	appGUIDLines, err := c.output("app", "--guid", appName)
+	if err != nil {
+		return "", err
+	}
+	appGUID := strings.TrimSpace(appGUIDLines)
+	if appGUID == "" {
+		return "", fmt.Errorf("Expected app ID for '%s' was empty.", appName)
+	}
+	return appGUID, nil
+}
+
+func (c *Cf) getAppEnv(appGUID string) (string, error) {
+	return c.output("curl", "/v2/apps/"+appGUID+"/env")
+}
+
+func (c *Cf) bindService(appName string, serviceInstanceName string) error {
+	return c.run("bind-service", appName, serviceInstanceName)
+}
+
+func (c *Cf) forceDeleteApp(appName string) error {
```

Comment:
I don't think this function needs the `force` in the name. The `-f` is merely an implementation detail of the non-interactive usage of the CLI.

Reply:
🙌